src/dso_api/dynamic_api/serializers.py

```diff
@@ -303,11 +303,31 @@ def get_fields(self):
         temporal = ds.temporal
         if temporal is not None and "identifier" in temporal:
             identifier_fields.append(temporal["identifier"])
+        capitalized_identifier_fields = [
+            identifier_field.capitalize() for identifier_field in identifier_fields
+        ]
+        fk_identifier_fields = []
+        for field_name, field in fields.items():
+            if isinstance(field, TemporalHyperlinkedRelatedField):
+                identifier_fields += [
+                    f"{field_name}{suffix}" for suffix in capitalized_identifier_fields
+                ]
+            if isinstance(field, TemporalReadOnlyField):
+                fk_identifier_fields.append(field_name)
+        for fk_identifier_field in fk_identifier_fields:
+            root_field_name = fk_identifier_field[:-2]  # without "Id"
+            for field_name, field in fields.items():
+                if field_name not in fk_identifier_fields and field_name.startswith(
+                    root_field_name
+                ):
+                    identifier_fields.append(field_name)
+
```

Comment:
Dit zou denk ik overzichtelijker zijn in losse functies (principes "Clean Code"):

identifier_fields += self._get_related_identifier_fields()

Reply:
De complexiteit van veld selectie ondergebracht in een aparte functie waardoor het geheel overzichterlijker wordt. Zelfde ook gedaan voor de get_fields in de DynamicLinksSerializer.